backend/server/routers/courses.py

```diff
@@ -444,6 +445,70 @@ def courses_unlocked_when_taken(userData: UserData, courseToBeTaken: str) -> Dic
         'indirect_unlock': sorted(list(indirect_unlock))
     }
 
+
+@router.get(
+    "/termsOffered/{course}",
+    response_model=TermsList,
+    responses={
+        400: {
+            "description": "The given course code could not be found in the database",
+        },
+        200: {
+            "description": "Returns the terms a course is offered in",
+            "content": {
+                "application/json": {
+                    "example": {
+                        "terms": ["21T2", "21T2", "21T3"],
+                    },
+                },
+            },
+        },
+    }
+)
+def terms_offered(course: str) -> Dict[str, List[str]]:
```

Comment:
We need to ask for the years they want, and give them the terms **for every year accurately**.
For example, 2041 was offered in 22t1. We need to help the frontend know this.

An example would be:
TermsOffered(course, sYear, lYear):
  Return {
     2020: [t1, T2],
     2021: [T3]
  }

Reply:
realised I misread the spec the moment I opened this :/